### DISCUSSION TITLE
Databricks Provider (Version 6.1.0) seems to bump the airflow version to latest

### DISCUSSION THREAD

**Discussion by Rutvijjs** [apache/airflow#40077] [Q&A] [answered]
### Apache Airflow Provider(s)

databricks

### Versions of Apache Airflow Providers

6.1.0 and above

### Apache Airflow version

2.6.3

### Operating System

Mac-os

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

Deployed into a kubernetes cluster through a docker image.

### What happened

We have been using the Databricks provider for quite some time with the version 6.1.0. However, recently we started running into an issue with our airflow install on a kubernetes cluster (through a dockerfile). Our docker file is using the 2.6.3 image of airflow yet when we do a pip install for databricks provider (for versions greater than 6.1.0 and even tested with just installing 6.1.0) it will automatically update the version of airflow to the latest (at this time this is 2.9.1) causing other dependency issues installed through the airflow image (2.6.3). 

### What you think should happen instead

The airflow image should not be updated to the latest version but instead be kept to what the version is specified in the image.

### How to reproduce

run a docker file that expands the apache-airflow 2.6.3 image and add a step for pip install. As follows:

```
FROM apache/airflow:2.6.3-python3.9

USER root

RUN apt update
RUN apt install openjdk-11-jre -y

USER airflow
RUN pip install --no-cache-dir \
        apache-airflow-providers-databricks==6.1.0
```

### Anything else

Every time i install using the versions

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by eladkal** [comment, marked as answer]
This is not about Airflow. It's about how pip works.
I recommand following the docs by doing so you will avoid such problems
https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html#managing-providers-separately-from-airflow-core

**Reply by Rutvijjs** [reply]
The problem i'm facing is that the way i'm running the dockerfile worked prior to end of may. It stopped working with the same configuration just recently, so i'm confused why now I need to change this. Also i'm pretty sure i'm following the doc minus adding a pip install for the apache-airflow specifically since i'm taking it from an apache-airflow image. Regardless, I will try with that.

**Reply by Rutvijjs** [reply]
Adding a pip install did solve this issue! Thank you for your suggestion. Still curious as to why this suddenly happened now.
